spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/MessageConverterConfigurer.java

```diff
@@ -125,13 +126,16 @@ private void configureMessageChannel(MessageChannel channel, String channelName,
 					getPartitionKeyExtractorStrategy(producerProperties),
 					getPartitionSelectorStrategy(producerProperties)));
 		}
-		if (input) {
-			messageChannel.addInterceptor(new InboundMessageConvertingInterceptor());
-		}
-		// TODO: Set all interceptors in the correct order for input/output channels
+		ConsumerProperties consumerProperties = bindingProperties.getConsumer();
 		if (StringUtils.hasText(contentType)) {
```

Comment:
Isn't this only for the case where the user explicitly sets the `contentType` in the binding properties for the channel name and also use the nativeDecoding for the decoding?

Reply:
@ilayaperumalg it doesn't matter if `contentType` is set on the channel or not. If native encoding/decoding is enabled, we will skip any message conversion. If we don't do that, based on the recent content type redesign, we could end up converting the message in certain cases.